model-engine/model_engine_server/domain/use_cases/llm_model_endpoint_use_cases.py

```diff
@@ -831,6 +831,54 @@ def deepspeed_result_to_tokens(result: Dict[str, Any]) -> List[TokenOutput]:
     return tokens
 
 
+def validate_and_update_completion_params(
+    inference_framework: LLMInferenceFramework,
+    request: Union[CompletionSyncV1Request, CompletionStreamV1Request],
```

Comment:
btw if the type checker is still giving you trouble maybe https://docs.python.org/3.8/library/typing.html#user-defined-generic-types would help? at least this feels like the "proper" way to do things to me

Reply:
I was able to pass the check if I call the function like this
```
new_request = validate_and_update_completion_params(endpoint_content.inference_framework, request)
assert isinstance(new_request, CompletionSyncV1Request)
request = new_request
```

Reply:
does something like `new_request: CompletionSyncV1Request = validate...` also work? basically another way of telling the type checker that you know it'll be a CompletionSyncV1Request

Reply:
Doesn't work... I think the problem is if the return type is union, will have to do some type narrowing https://mypy.readthedocs.io/en/stable/type_narrowing.html

Reply:
you can use generics https://mypy.readthedocs.io/en/stable/generics.html

```
from typing import TypeVar, Sequence

T = TypeVar('T')

def validate_and_update_completion_params(
    inference_framework: LLMInferenceFramework,
    request: T,
):
...
```